### PR TITLE
test: certificate fixes for fedora

### PR DESCRIPTION
**Pull request by heyitsanthony**
fedora doesn't support ECDSA and darwin's curl wants pkcs12 certs

### CODE REVIEW

File: e2e/v2_curl_test.go [hunk 1]
```diff
@@ -139,10 +139,10 @@ func cURLPrefixArgs(clus *etcdProcessCluster, method string, req cURLReq) []stri
 		if clus.cfg.clientTLS != clientTLSAndNonTLS {
 			panic("should not use cURLPrefixArgsUseTLS when serving only TLS or non-TLS")
 		}
-		cmdArgs = append(cmdArgs, "--cacert", caPath, "--cert", certPath, "--key", privateKeyPath)
+		cmdArgs = append(cmdArgs, "--cacert", caPath, "--cert", pkcs12Path+":123", "--cert-type", "P12")
```
Comment (gyuho) [inline e2e/v2_curl_test.go]: What is `+":123"`?

Comment (heyitsanthony) [inline e2e/v2_curl_test.go]: password; I tried making it with a blank password but curl still complained asking for one

Comment (gyuho) [inline e2e/v2_curl_test.go]: Ok got it. Thanks.

**Comment by gyuho** [comment]
lgtm. Defer to @xiang90

**Comment by heyitsanthony** [comment]
nuking the P12 cert commit since CI doesn't support it

**Comment by xiang90** [comment]
lgtm

**Comment by heyitsanthony** [comment]
holding off on merge until I can test on a fedora vm tonight

**Comment by heyitsanthony** [comment]
tested and works, merging

**Pull request merged**